### PR TITLE
Support creating Scalars/ColumnVectors from utf8 strings directly.

### PR DESCRIPTION
**Pull request by firestarman**
Update to create Scalars/ColumnVectors from utf8 strings directly, since the issue https://github.com/rapidsai/cudf/issues/8137 has been solved.

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala [hunk 1]
```diff
@@ -110,9 +110,7 @@ object GpuScalar extends Arm with Logging {
   /** Converts an element for nested lists */
   private def convertElementTo(element: Any, elementType: DataType): Any = elementType match {
     case _ if element == null => null
-    // StructData does not support utf8 string yet, so parses it here instead of calling the
-    //  `convertUTF8StringTo`. Tracked by https://github.com/rapidsai/cudf/issues/8137
-    case StringType => element.asInstanceOf[UTF8String].toString
+    case StringType => element.asInstanceOf[UTF8String].getBytes
```
Comment (wjxiz1992) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala:113]: Only StringType are representable by Bytes? Bytes cannot be casted to other types? for example an IntegerType?

Comment (firestarman) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala:113]: Here is only for `StringType`, and it is array of byte inside an `UTF8String`.

**Review by wjxiz1992 — APPROVED** [review]
LGTM

**Pull request merged**